File: 14719.py
import sys

n,m = map(int, sys.stdin.readline().split())

arr = list(map(int, sys.stdin.readline().split()))

ans = 0

for i in range(1, len(arr)-1):
    l_max = max(arr[:i+1])
    r_max = max(arr[i:])

    ans += min(l_max, r_max) - arr[i]

print(ans)

File: 14719.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <queue>

using namespace std;

int main(){
    int n, m;
    cin >> n >> m;

    vector<int> a(m);

    for(int i=0; i<m; i++){
        cin>>a[i];
    }

    int ans = 0;

    for(int i=1; i<m-1; i++){
        int left_max = 0, right_max = 0;

        for(int j=0; j<i; j++){
            right_max = max(right_max, a[j]);
        }
        
        for(int j=i+1; j<m; j++){
            left_max = max(left_max, a[j]);
        }
        int diff = min(left_max, right_max) - a[i];
        if (diff>0){
            ans += diff;  
        }
    }

    cout << ans;
}